lib/maths/CBoostedTreeImpl.cc

```diff
@@ -341,6 +343,11 @@ std::size_t CBoostedTreeImpl::estimateMemoryUsage(std::size_t numberRows,
     std::size_t hyperparametersMemoryUsage{numberColumns * sizeof(double)};
     std::size_t tunableHyperparametersMemoryUsage{m_TunableHyperparameters.size() *
                                                   sizeof(int)};
+    std::size_t hyperparameterSamplesMemoryUsage{
+        m_HyperparameterSamples.empty()
+            ? 0
+            : m_HyperparameterSamples.size() *
+                  m_HyperparameterSamples[0].size() * sizeof(double)};
```

Comment:
This function gets called before train so I don't think this works. I'd propose:
```
    std::size_t hyperparameterSamplesMemoryUsage{
        (m_NumberRounds / 3 + 1) * m_TunableHyperparameters.size() * sizeof(double)};
```
I'd be inclined to pull `m_NumberRounds / 3 + 1` into a function as well so this properly encapsulated.

Reply:
Except that I think this should also be using `this->numberHyperparametersToTune()` since `m_TunableHyperparameters.size()` also won't be initialised at the point this function is called.

Reply:
:+1: Thank you for the suggestion. Fixed now.